clients/src/test/java/org/apache/kafka/common/network/SelectorTest.java

```diff
@@ -580,6 +582,41 @@ public void testConnectDisconnectDuringInSinglePoll() throws Exception {
         control.verify();
     }
 
+    @Test
+    public void testOutboundConnectionsCountInConnectionCreationMetric() throws Exception {
+        // create connections
+        int conns = 5;
+        InetSocketAddress addr = new InetSocketAddress("localhost", server.port);
+        for (int i = 0; i < conns; i++)
+            connect(Integer.toString(i), addr);
+
+        selector.poll(0L);
+
+        assertEquals(getMetric("connection-creation-total").metricValue(), (double) conns);
+    }
+
+    @Test
+    public void testInboundConnectionsCountInConnectionCreationMetric() throws Exception {
+        int conns = 5;
+
+        try (ServerSocketChannel ss = ServerSocketChannel.open()) {
+            ss.bind(new InetSocketAddress(0));
+            InetSocketAddress serverAddress = (InetSocketAddress) ss.getLocalAddress();
+
+            for (int i = 0; i < conns; i++) {
+                Thread sender = createSender(serverAddress, randomPayload(1));
+                sender.start();
+                sender.join(1000);
+                SocketChannel channel = ss.accept();
+                channel.configureBlocking(false);
+
+                selector.register(Integer.toString(i), channel);
```

Comment:
Not related to your change, but perhaps we could also verify `connection-count` metric? This is a gauge, and should also equal `conns` in both the tests after the connections are established. What do you think?

Reply:
Doesn't hurt to guarantee consistency with both metrics imo, so yeah, good suggestion